ste/uploader-blobFS.go

```diff
@@ -25,13 +25,56 @@ import (
 	"fmt"
 	"net/url"
 	"os"
+	"sync"
 	"time"
 
 	"github.com/Azure/azure-pipeline-go/pipeline"
 	"github.com/Azure/azure-storage-azcopy/azbfs"
 	"github.com/Azure/azure-storage-azcopy/common"
 )
 
+const bfsTrackerThreshold = (1024 ^ 3) * 10 // 10 GB
+
+var bfsTrackerOnce sync.Once
+var bfsTrackerTodo = make(chan *blobFSUploader, 5000)
+
+/*
+	TODO: Find a better way to init the channel
+
+	Currently, 5000 tracker entries may fill up and cause transfer upload funcs to end up blocking until they can push.
+	This *may* not actually degrade perf because we're trying to upload multiple chunks at a time
+	However, I also know that we do a limited number of goroutines (I think).
+*/
+
+func bfsTrackerRoutine() {
+	for bfsu := range bfsTrackerTodo { //The routine will sleep when we try to pull from the range, letting others do work
+		for ; bfsu.ti < len(bfsu.tp) && bfsu.tp[bfsu.ti].Completed; bfsu.ti++ {
+			if bfsu.ti == -1 {
+				continue
+			}
+
+			id := bfsu.tp[bfsu.ti].ID
+			if id.OffsetInFile-bfsu.lf >= bfsTrackerThreshold || bfsu.ti+1 == len(bfsu.tp) {
```

Comment:
As far as I can tell, this doesn't handle the case where the last chunk completes before, say, the second-to-last.  But... if we reinstate the code that runs in the Epilogue, then that's OK, because the epilogue can do a final flush

Reply:
Oops, no I was mistaken on the previous comment re last chunk completing before second to last.. Will delete it .